paparazzi/paparazzi-annotation/README.md

```diff
@@ -0,0 +1,161 @@
+  MyView(title = "Hello, Annotation")
+}
+```
+
+The annotation exposes a [number of parameters](./src/main/java/app/cash/paparazzi/annotation/api/Paparazzi.kt) to configure Paparazzi as you see fit.
+
+```kotlin
+@Paparazzi(
+  name = "big text pixel scroll",
+  fontScale = 3.0f,
+  deviceConfig = DeviceConfig.PIXEL_5,
+  renderingMode = RenderingMode.V_SCROLL,
+  theme = "android:Theme.Material.Fullscreen",
+)
+```
+
+## Composable Previews
+If you're already using preview functions to visualize your composable UI, then you can simply annotate that function to create a test.
+
+```kotlin
+@Paparazzi
+@Preview
+@Composable
+fun MyViewPreview() {
+  MyView(title = "Hello, Annotation")
+}
+```
+
+*A word of caution about this approach: Previews are meant as a developer tool and can change to help visualize different scenarios. Changing a preview will likely invalidate a test and could create confusion.*
+
+## Annotation Composition
+If you want to create more than one test class for a composable function, or have a highly-configured annotation that you want to re-use on multiple functions, then you can define a custom annotation as a fixture to hold your `@Paparazzi` definitions. Each annotation creates a separate test class.
+
+```kotlin
+@Paparazzi(
+  name = "pixel5",
+  deviceConfig = DeviceConfig.PIXEL_5,
+)
+@Paparazzi(
+  name = "large text",
+  fontScale = 2.0f,
+)
+annotation class MyPaparazzi
+
+@MyPaparazzi
+@Composable
+fun MyViewTest() {
+  MyView(title = "Hello Paparazzi, in multiple ways")
+}
+```
+This style of composition can even be nested, if desired.
+
+```kotlin
+@Paparazzi(
+  name = "pixel5",
+  deviceConfig = DeviceConfig.PIXEL_5,
+)
+@Paparazzi(
+  name = "large text",
+  fontScale = 2.0f,
+)
+annotation class MyPaparazzi
+
+@Paparazzi(
+  name = "landscape",
+  orientation = ScreenOrientation.LANDSCAPE,
+)
+@MyPaparazzi
+annotation class MyAlsoLandscapePaparazzi
+```
+
+## Composable Wrapping
+It might be necessary for you to wrap your composable with additional logic to prepare it for the Paparazzi test. To do this, implement the provided `ComposableWrapper` interface.
+
+```kotlin
+class BlueBoxComposableWrapper : ComposableWrapper {
+  @Composable
+  override fun wrap(content: @Composable () -> Unit) {
+    Box(
+      modifier = Modifier
+        .wrapContentSize()
+        .background(Color.Blue)
+        .padding(24.dp)
+    ) {
+      content()
+    }
+  }
+}
+
+@Paparazzi(
+  name = "blue box",
+  composableWrapper = BlueBoxComposableWrapper::class
+)
+```
+
+## Test Parameter Injection
+The `@Paparazzi` annotation also leverages `TestParameterInjector` to execute multiple tests on a handful of configuration parameters.
+
+### Font Scaling
+The annotation exposes the `fontScales` configuration which accepts an array of `Float` values. Setting this parameter will override the value set in the (non-plural) `fontScale` parameter.
+
+```kotlin
+@Paparazzi(
+  name = "fontScaling",
+  fontScales = [1.0f, 2.0f, 3.0f]
```

Comment:
(These sample values are dangerous; should be 1.0, 1.1, 1.2 or something. 3x font scale is a nightmare)